onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py

```diff
@@ -636,8 +876,9 @@ def _process_subgraph(self, graph_stack: list[GraphProto]):
             if node.name in self.nodes_to_exclude:
                 logger.info(f"exclude to quantize {node.name} as specified by nodes_to_exclude...")
                 out_nodes = [node]
-            elif self.algo_config is not None and self.algo_config.algorithm == "HQQ":
-                out_nodes = self.node_quantizer.quantize(node, graph_stack)
+            elif self.nodes_to_include and (node.name not in self.nodes_to_include):
```

Comment:
>     elif self.nodes_to_include and (node.name not in self.nodes_to_include): [](http://example.com/codeflow?start=7&length=77)

node_to_include should have priority over type_to_quantize, i.e., if Gather is not in type_to_quantize but node_to_incude includes a Gather to quantize, we should still quantize it.

Reply:
updated the logic